sdk-python/omnilogger/input_validation.py

```diff
@@ -1,27 +1,47 @@
 import datetime
 
+from prisma.types import llm_logsCreateInput
+
+from omnilogger.errors import LogDictKeyError
+
 
 def check_url_type(url):
     if not isinstance(url, str):
         raise TypeError("Have you checked your url connection string is correct?")
 
 
-def check_log_type(log):
-    if not isinstance(log, dict):
-        raise TypeError(
-            "Have you checked your log is a dictionary?\nlog must be a dictionary with"
-            " keys: datetime_utc, input, output, total_tokens"
-        )
-    if set(log.keys()) != {"datetime_utc", "input", "output", "total_tokens"}:
-        raise KeyError(
-            "log must be a dictionary with keys: datetime_utc, input, output,"
-            " total_tokens"
-        )
-    if not isinstance(log["datetime_utc"], datetime.datetime):
-        raise TypeError("log.datetime_utc must be a datetime object")
-    if not isinstance(log["input"], str):
-        raise TypeError("log.input must be a string")
-    if not isinstance(log["output"], str):
-        raise TypeError("log.output must be a string")
-    if type(log["total_tokens"]) is not int:
-        raise TypeError("log.total_tokens must be an integer")
+def check_log_type(log: llm_logsCreateInput):
+    assert isinstance(log, dict), LogDictKeyError()
+    assert set({**log.copy(), "id": -1}.keys()) == set(
+        llm_logsCreateInput.__annotations__.keys()
+    ), LogDictKeyError()
+
+    assert isinstance(log["datetime_utc"], datetime.datetime), TypeError(
+        "log.datetime_utc must be a datetime object"
+    )
+    assert isinstance(log["input_string"], str), TypeError(
+        "log.input_string must be a string"
+    )
+    assert isinstance(log["output_string"], str), TypeError(
+        "log.output_string must be a string"
+    )
+    assert type(log["total_tokens"]) is int, TypeError(
+        "log.total_tokens must be an integer"
+    )
+
+
+if __name__ == "__main__":
+    log = llm_logsCreateInput(
+        datetime_utc=datetime.datetime.utcnow(),
+        input_string="What is this?",
+        output_string="This is a test log",
+        total_tokens=5,
+    )
+    mockLogWithId = {**log.copy(), "id": -1}
+    mockLogWithId["id"] = -1
+    print(set(log.keys()))
+    print(set(llm_logsCreateInput.__annotations__.keys()))
+    print(
+        set({**log.copy(), "id": -1}.keys())
+        == set(llm_logsCreateInput.__annotations__.keys())
+    )
```

Comment:
Is this to do testing? Is it run with pytest?

Reply:
this should not be here